components/registry-facade/pkg/registry/layersource.go

```diff
@@ -459,6 +462,130 @@ func (src *SpecMappedImagedSource) getDelegate(ctx context.Context, spec *api.Im
 	return lsrc, nil
 }
 
+// NewDynamicSpecMappedImageSource creates a new dynamic spec mapped image source
+func NewDynamicSpecMappedImageSource(resolver ResolverProvider, refsSource RefsSource) (*SpecDynamicMappedImagedSource, error) {
```

Comment:
How does the `DynamicSpecMappedImageSource` differ from the `SpecMappedImageSource`?